dev-portal/src/pages/Admin/Accounts/RegisteredAccounts.jsx

```diff
@@ -0,0 +1,245 @@
+      />
+      <PromoteAccountModal
+        account={selectedAccount}
+        onConfirm={onConfirmPromote}
+        open={promoteModalOpen}
+        onClose={() => setPromoteModalOpen(false)}
+      />
+    </Container>
+  )
+}
+export default RegisteredAccounts
+
+const TableActions = React.memo(
+  ({ canDelete, onClickDelete, canPromote, onClickPromote }) => (
+    <Button.Group>
+      <Button content='Delete' disabled={!canDelete} onClick={onClickDelete} />
+      <Button
+        content='Promote to Admin'
+        disabled={!canPromote}
+        onClick={onClickPromote}
+      />
+    </Button.Group>
+  ),
+)
+
+const DeleteAccountModal = React.memo(
+  ({ account, onConfirm, open, onClose }) =>
+    account && (
+      <Modal size='small' open={open} onClose={onClose}>
+        <Modal.Header>Confirm deletion</Modal.Header>
+        <Modal.Content>
+          <p>
+            Are you sure you want to delete the account{' '}
+            <strong>{account.emailAddress}</strong>, and de-activate the
+            associated API key? This action is irreversible.
+          </p>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button negative onClick={onConfirm}>
+            Delete
+          </Button>
+        </Modal.Actions>
+      </Modal>
+    ),
+)
+
+const PromoteAccountModal = React.memo(
+  ({ account, onConfirm, open, onClose }) =>
+    account && (
+      <Modal size='small' open={open} onClose={onClose}>
+        <Modal.Header>Confirm promotion</Modal.Header>
+        <Modal.Content>
+          <p>
+            Are you sure you want to promote the account{' '}
+            <strong>{account.emailAddress}</strong> to Admin? This will allow
+            the account to perform any Admin actions, including deleting and
+            promoting other accounts.
+          </p>
+          <p>
+            Only the owner of the Developer Portal can demote the account,
+            through the Cognito console.
+          </p>
+        </Modal.Content>
+        <Modal.Actions>
+          <Button onClick={onClose}>Cancel</Button>
+          <Button negative onClick={onConfirm}>
+            Promote
+          </Button>
+        </Modal.Actions>
+      </Modal>
+    ),
+)
+
+const MESSAGE_RENDERERS = {
+  [DELETE_SUCCESS]: ({ account }, onDismiss) => (
+    <DeleteSuccessMessage account={account} onDismiss={onDismiss} />
+  ),
+  [DELETE_FAILURE]: ({ account, errorMessage }, onDismiss) => (
+    <DeleteFailureMessage
+      account={account}
+      errorMessage={errorMessage}
+      onDismiss={onDismiss}
+    />
+  ),
+  [PROMOTE_SUCCESS]: ({ account }, onDismiss) => (
+    <PromoteSuccessMessage account={account} onDismiss={onDismiss} />
+  ),
+  [PROMOTE_FAILURE]: ({ account, errorMessage }, onDismiss) => (
+    <PromoteFailureMessage
+      account={account}
+      errorMessage={errorMessage}
+      onDismiss={onDismiss}
+    />
+  ),
+}
```

Comment:
The whole messages thing seems really complex. Why not just have sendMessage accept a JSX element or React component and just render that?

Reply:
I think it is more complex than necessary. I wanted the "dismissing" functionality to be handled by `MessageList`, but I think I've found a simpler way to do so. The simpler way will be very close to "have `sendMessage` accept a JSX element and just render that", but additionally accept a `dismiss` callback which will dismiss the message when called.